analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -29,5 +29,81 @@ public class PublicMethodArgumentsShouldBeCheckedForNull : SymbolicRuleCheck
 
     protected override DiagnosticDescriptor Rule => S3900;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute()
+    {
+        return IsAccessibleFromOtherAssemblies(Node)
+            && (IsRelevantMethod(Node) || IsRelevantPropertyAccessor(Node));
+
+        static bool IsRelevantMethod(SyntaxNode node) =>
+            node is BaseMethodDeclarationSyntax { ParameterList.Parameters.Count: > 0 } method
```

Comment:
Do we have exceptions regarding `this` parameters (extension methods)? Do we need to take this into account here?

Reply:
If all parameters are `out` parameters, ShouldExecute should return false.

Reply:
I initially added the common base class, [but got a comment for it](https://github.com/SonarSource/sonar-dotnet/pull/6917#discussion_r1135079895). So it will be added later with the VB.NET implementation.

Invocation of an extension method will not raise an issue. However, this will not be dealt with in this PR, as `ShouldExecute()` only operates on the syntax level and can't check whether a method call is referring to an extension method.

I added a filter for `out` params.